src/text/word2vec-utils.ts

```diff
@@ -0,0 +1,90 @@
+export function clean(text: string, stopwords: string[] | undefined): Array<string> {
+  let textArray = text.split(' ');
+  //filter out empty string
+  textArray = textArray.filter((value) => {
+    return value != '';
+  });
+
+  //filter out stopwords.
+  if (stopwords) {
+    textArray = textArray.filter((value) => {
+      return !stopwords.includes(value);
+    });
+  }
+  //remove punctuation
+  textArray = textArray.map((value) => {
+    return value.replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g, "");
+  });
```

Comment:
The function takes 3-times to finish indexing. IMO, it is possible to minimize it into one round

Reply:
Yes. looking at readability at first, but that should be possible even when minimized.